src/Jenssegers/Mongodb/Concerns/TransactionManager.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Jenssegers\Mongodb\Concerns;
+
+use Closure;
+use Exception;
+use MongoDB\Driver\Session;
+
+use function MongoDB\with_transaction;
+
+trait TransactionManager
+{
+    /**
+     * create a session and start a transaction in session
+     *
+     * In version 4.0, MongoDB supports multi-document transactions on replica sets.
+     * In version 4.2, MongoDB introduces distributed transactions, which adds support for multi-document transactions on sharded clusters and incorporates the existing support for multi-document transactions on replica sets.
+     * To use transactions on MongoDB 4.2 deployments(replica sets and sharded clusters), clients must use MongoDB drivers updated for MongoDB 4.2.
+     *
+     * @see https://docs.mongodb.com/manual/core/transactions/
+     * @return void
+     */
+    public function beginTransaction(?array $options = [])
+    {
+        $session = $this->getSession();
+
+        if (!$session) {
+            $this->session_key = uniqid();
+            $session = $this->connection->startSession();
+            $this->sessions[$this->session_key] = $session;
```

Comment:
Since we're only managing a single session, we can do away with the array and store the single session directly. This also makes the session key unnecessary.